src/std_extensions/arithmetic/int_types.rs

```diff
@@ -30,37 +29,22 @@ fn int_custom_type(n: u8) -> CustomType {
 /// Integer type of a given bit width.
 /// Depending on the operation, the semantic interpretation may be unsigned integer, signed integer
 /// or bit string.
-pub fn int_type(n: u8) -> Type {
-    Type::new_extension(int_custom_type(n))
+pub fn int_type(width_power: u8) -> Type {
+    Type::new_extension(int_custom_type(width_power))
 }
 
-fn is_valid_width(n: u8) -> bool {
-    (n == 1)
-        || (n == 2)
-        || (n == 4)
-        || (n == 8)
-        || (n == 16)
-        || (n == 32)
-        || (n == 64)
-        || (n == 128)
+const fn is_valid_width(n: u8) -> bool {
+    (n <= (1u8 << (POWERS_OF_TWO - 1))) && ((n & (n - 1)) == 0)
 }
 
+const POWERS_OF_TWO: usize = 8;
+
 /// Get the bit width of the specified integer type, or error if the width is not supported.
-pub fn get_width(arg: &TypeArg) -> Result<u8, SignatureError> {
-    let n: u8 = match arg {
-        TypeArg::USize(n) => *n as u8,
-        _ => {
-            return Err(TypeArgError::TypeMismatch {
-                arg: arg.clone(),
-                param: TypeParam::USize,
-            }
-            .into());
-        }
-    };
-    if !is_valid_width(n) {
-        return Err(TypeArgError::InvalidValue(arg.clone()).into());
+pub fn get_width_power(arg: &TypeArg) -> u8 {
+    match arg {
+        TypeArg::SimplePredicate(n) if *n < POWERS_OF_TWO => *n as u8,
+        _ => panic!("type check should prevent this."),
```

Comment:
Should this really be a panic? The tests demonstrate that it's possible to invoke via the public API, I think?

Reply:
yes - I agree with alan that this function shouldn't be in the public API